### PR TITLE
Fix Hugging Face server EncoderModel not returning probabilities by correctly passing --return_probabilities flag  (#3958)

### LINKED ISSUE #3958: Hugging Face server not returning probability scores with --return_probabilities flag
/kind bug

**What steps did you take and what happened:**
I deployed an InferenceService using Hugging Face server with a Hugging Face model for sequence classification.
In the InferenceService YAML, I used the --return_probabilities argument, expecting the model to return all probability distributions for the predictions. However, the model is only returning the predicted class label, not the probability scores.
After reviewing the code, it seems the return_probabilities argument is not being passed correctly to the HuggingfaceEncoderModel. Specifically, the model initialization does not include this parameter, so the probabilities are not being returned as expected.

**What did you expect to happen:**
I expected the InferenceService to return the complete probability distribution for each prediction, as indicated by the --return_probabilities argument.

**What's the InferenceService yaml:**
```
apiVersion: serving.kserve.io/v1beta1
kind: InferenceService
metadata:
  name: huggingface-test
spec:
  predictor:
    model:
      modelFormat:
        name: huggingface
      args:
      - --model_id=XSY/albert-base-v2-fakenews-discriminator
      - --backend=huggingface
      - --task=sequence_classification
      - --return_probabilities
      resources:
        limits:
          cpu: "1"
          memory: 2Gi
        requests:
          cpu: 100m
          memory: 2Gi
```

**Anything else you would like to add:**
I use 
```
curl -v -H "Host: huggingface-test.default.example.com" -H "content-type:application/json"  http://localhost:8080/v1/models/huggingface-test:predict  -d '{"instances": ["hello"]}'
```
return
```
* Connection #0 to host localhost left intact
{"predictions":[0]}%
```

It seems the HuggingfaceEncoderModel class does not pass the return_probabilities argument correctly. Although its action="store_true" should make it True, the model's initialization does not include this parameter, which prevents the probability distributions from being returned as expected.
The argument is defined as:
```
parser.add_argument(
    "--return_probabilities",
    action="store_true",
    help="Return all probabilities",
)
```
But in the model initialization:
```
model = HuggingfaceEncoderModel(
                model_name=args.model_name,
                model_id_or_path=model_id_or_path,
                task=task,
                model_config=model_config,
                model_revision=kwargs.get("model_revision", None),
                tokenizer_revision=kwargs.get("tokenizer_revision", None),
                do_lower_case=not kwargs.get("disable_lower_case", False),
                add_special_tokens=not kwargs.get("disable_special_tokens", False),
                max_length=kwargs["max_model_len"],
                dtype=dtype,
                trust_remote_code=kwargs["trust_remote_code"],
                tensor_input_names=kwargs.get("tensor_input_names", None),
                return_token_type_ids=kwargs.get("return_token_type_ids", None),
                predictor_config=predictor_config,
                request_logger=request_logger,
            )
```
In Encoder
```
def __init__(
        self,
        return_probabilities: bool = False,
        ...
):
```

**Environment:**

- Istio Version:1.20.4
- Knative Version:1.13.1
- KServe Version: master
- Kubeflow version:
- Cloud Environment:[k8s_istio/istio_dex/gcp_basic_auth/gcp_iap/aws/aws_cognito/ibm] : no
- minikube version: v1.34.0
- Kubernetes version: (use `kubectl version`):v1.31.0
- OS (e.g. from `/etc/os-release`):Ubuntu 24.04 LTS

### PR DESCRIPTION
**Pull request by oplushappy**
**What this PR does / why we need it**:
Fixes an issue with missing probability distributions in predictions when using the Hugging Face server EncoderModel with `--return_probabilities`.
Ensures `--return_probabilities` is correctly passed, enabling the model to return full probability scores instead of just the predicted class label.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #3958

**Type of changes**

- [x] Bug fix (non-breaking change which fixes an issue)

**Feature/Issue validation/testing**:

- [x]  Added `test_huggingface_v2_sequence_classification_with_probabilities` in `test_huggingface.py`
- [x]  Verified that existing functionality is unaffected by the change

**Checklist**:

- [x] Have you added unit/e2e tests that prove your fix is effective or that this feature works?

**Release note**:

```release-note
Fixed missing probability distributions in Huggingface server EncoderModel when --return_probabilities is specified.
```

### CODE REVIEW

File: test/e2e/predictor/test_huggingface.py [hunk 1]
```diff
@@ -339,3 +339,57 @@ async def test_huggingface_v2_text_embedding(rest_v2_client):
...
+    )
+    kserve_client.create(isvc)
+    kserve_client.wait_isvc_ready(service_name, namespace=KSERVE_TEST_NAMESPACE)
+
+    res = await predict_isvc(
+        rest_v2_client,
+        service_name,
+        "./data/bert_sequence_classification_v2.json",
+    )
+    assert res.outputs[0].data == ['{0: -2.152204, 1: 2.5094059}']
```
Comment (sivanantha321) [inline test/e2e/predictor/test_huggingface.py]: You have to limit the precision like it is done here https://github.com/kserve/kserve/blob/master/python/huggingfaceserver/tests/test_output.py

**Comment by sivanantha321** [comment]
@oplushappy Can you help fix the lint errors ?

**Comment by oplushappy** [comment]
> @oplushappy Can you help fix the lint errors ?

Yes, I can.

**Comment by sivanantha321** [comment]
@oplushappy Can you help sign the commit ? The DCO check is failing.

**Comment by oplushappy** [comment]
> @oplushappy Can you help sign the commit ? The DCO check is failing.

Yes, I finished it.

**Comment by yuzisun** [comment]
@oplushappy can you help fix the linting errors?

**Comment by oplushappy** [comment]
> CO

**Comment by oplushappy** [comment]
> @oplushappy can you help fix the linting errors?

Yes, I finished it.

**Comment by sivanantha321** [comment]
@oplushappy Hi, Can you do a rebase from master ? There is a fix for failing CI. Thanks !

**Comment by oplushappy** [comment]
> @oplushappy Hi, Can you do a rebase from master ? There is a fix for failing CI. Thanks !

Yes, I finished rebasing from master.

**Pull request merged**